src/main/java/org/fenixedu/academic/service/services/manager/MergeExecutionCourses.java

```diff
@@ -424,6 +426,14 @@ private static void copyPersistentGroups(ExecutionCourse executionCourseFrom, Ex
         }
     }
 
+    private static void copySenderMessages(ExecutionCourse executionCourseFrom, ExecutionCourse executionCourseTo) {
+        if (executionCourseFrom.getSender() != null) {
+            ExecutionCourseSender courseSenderTo = ExecutionCourseSender.newInstance(executionCourseTo);
+            courseSenderTo.getMessagesSet().addAll(executionCourseFrom.getSender().getMessagesSet());
+            executionCourseFrom.getSender().delete();
```

Comment:
The recipients that exist in an execution course are computed based on the execution course itself so there is no need to copy the recipients, it is only necessary to copy the messages that were sent.